api/src/main/java/com/spectralogic/tpfr/api/ServerServiceFactoryImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.spectralogic.tpfr.api;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.logging.HttpLoggingInterceptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import retrofit2.Retrofit;
+import retrofit2.converter.simplexml.SimpleXmlConverterFactory;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static okhttp3.OkHttpClient.Builder;
+
+public class ServerServiceFactoryImpl implements ServerServiceFactory {
+
+    private final Logger LOG = LoggerFactory.getLogger(ServerServiceFactoryImpl.class);
+    private ServerService serverService;
+
+    @Override
+    public Optional<ServerService> getServerService() {
+        return Optional.ofNullable(serverService);
+    }
+
+    @Override
+    public ServerService createServerService(final String endpoint) {
```

Comment:
The more I think about this the less I like the fact that I cannot get an instance of the API class back.  I realize part of the point of the ServerService class is to hide the fact that we're using retrofit under the hood, but if we expose the interface with the retrofit calls, then that gives the caller the option to make the rest calls synchronously, or asynchronously.